tools/sqlite3_api_wrapper/sqlite3_udf_api/cast_sqlite.cpp

```diff
@@ -0,0 +1,210 @@
+#include "cast_sqlite.hpp"
+
+#include "duckdb/common/vector_operations/vector_operations.hpp"
+#include "duckdb/common/types/validity_mask.hpp"
+
+bool CastSQLite::IsCastToVarchar(LogicalType type) {
+	LogicalTypeId type_id = type.id();
+	switch (type_id) {
+	case LogicalTypeId::TINYINT:
+	case LogicalTypeId::SMALLINT:
+	case LogicalTypeId::INTEGER:
+	case LogicalTypeId::BIGINT:
+	case LogicalTypeId::FLOAT:
+	case LogicalTypeId::DOUBLE:
+	case LogicalTypeId::CHAR:
+	case LogicalTypeId::VARCHAR:
+	case LogicalTypeId::BLOB:
+	case LogicalTypeId::SQLNULL:
+		return false; // supportted types
+	default:
+		return true; // types need casting to varchar
+	}
+}
+
+void CastSQLite::InputVectorsToVarchar(DataChunk &data_chunk) {
+	for (idx_t i = 0; i < data_chunk.ColumnCount(); ++i) {
+		if (CastSQLite::IsCastToVarchar(data_chunk.data[i].GetType())) {
+			LogicalType vchar_type(LogicalTypeId::VARCHAR);
+			Vector varchar_vector(vchar_type);
+
+			VectorOperations::Cast(data_chunk.data[i], varchar_vector, data_chunk.size(), true);
+
+			bool is_constant_vector = false;
+			if (data_chunk.data[i].GetVectorType() == VectorType::CONSTANT_VECTOR) {
+				is_constant_vector = true;
+				// it's needed to change to FLAT_VECTOR for the COPY works
+				data_chunk.data[i].SetVectorType(VectorType::FLAT_VECTOR);
+			}
+
+			data_chunk.data[i].SetType(vchar_type);
+			VectorOperations::Copy(varchar_vector, data_chunk.data[i], data_chunk.size(), 0, 0);
```

Comment:
Instead of both casting and copying, could we create a DataChunk that either (1) performs a cast if required, or (2) references the input vector of the other data chunk if not required? That would save a copy and simplify the code here somewhat.